packages/office-ui-fabric-react/src/components/Layer/Layer.base.tsx

```diff
@@ -23,7 +23,7 @@ let _defaultHostSelector: string | undefined;
 
 const getClassNames = classNamesFunction<ILayerStyleProps, ILayerStyles>();
 
-@customizable('Layer', ['theme'])
+@customizable('Layer', ['theme', 'hostId'])
```

Comment:
I really like this! But small question; before investing in `hostId`, should `ILayerProps.hostId` be `target` to conform to callout and other things that target entities?

I'd love to have consistency here, and if you want to do the layer targetting stuff, we should use this as an opportunity to invest in a consistent approach.

Reply:
What I mean is `target='#id'` or `target=[data-attribute]` or `target={element}` kind of thing using `querySelector`.

I am not opposed to this change, it makes sense to me and is not going to add complexity or weight but would like to just question the naming here.

Reply:
If we changed `hostId` to something else, wouldn't that be a breaking change?
One thing I wish is that we could do something more like this:

```ts
const myLayerChannel = createLayerChannel(); // Think React.createContext()

function render(): {
    return (
        <>
            <Customizer settings={ { 'Layer': { 'channel': myLayerChanne;; } } }>
                ... // Stuff that renders layers.
            </Customizer>
            <LayerHost channel={ myLayerChannel } />
        </>
    );
}
```
The problem with `id` is that since it's a string, you can't guarantee its uniqueness within a larger application. A wrapper like `createLayerChannel` could create an `id` for you internally and ensure that it's essentially 'private'.

Regarding `hostId`, that's the current field of `Layer`, so if we wanted something different wouldn't we need to add a new prop to `Layer` that `hostId` also edits?

Reply:
Looking deeper into the code, it's pretty coupled to `hostId` for the moment. I think it's best to do any major interface updates in a dedicated revision + breaking change. Plus, we should switch to use `React.createPortal` while we're at it.